validate/src/message.rs

```diff
@@ -130,6 +150,33 @@ pub enum MessageValidationErrorType {
     TooManyEmbeds,
 }
 
+/// Ensure an attachment's filename is correct.
+///
+/// The filename can contain ASCII alphanumeric characters, dots, dashes, and
+/// underscores.
+///
+/// # Errors
+///
+/// Returns an error of type [`AttachmentFilename`] if the filename is invalid.
+///
+/// [`AttachmentFilename`]: MessageValidationErrorType::AttachmentFilename
+pub fn attachment_filename(filename: impl AsRef<str>) -> Result<(), MessageValidationError> {
+    if filename
+        .as_ref()
+        .chars()
+        .any(|c| !(c.is_ascii_alphanumeric() || c == DOT || c == DASH || c == UNDERSCORE))
+    {
+        Err(MessageValidationError {
+            kind: MessageValidationErrorType::AttachmentFilename {
+                filename: filename.as_ref().to_string(),
+            },
+            source: None,
+        })
+    } else {
+        Ok(())
+    }
```

Comment:
LGTM

Just a personal touch on my end: I think checking whether it is valid instead of it being invalid is easier to understand.
```suggestion
    if filename
        .as_ref()
        .chars()
        .all(|c| c.is_ascii_alphanumeric() || c == DOT || c == DASH || c == UNDERSCORE)
    {
        Ok(())
    } else {
        Err(MessageValidationError {
            kind: MessageValidationErrorType::AttachmentFilename {
                filename: filename.as_ref().to_string(),
            },
            source: None,
        })
    }
```

Reply:
ah i was unaware of all. ill do that